packages/yourloops/components/device/utils/device.utils.ts

```diff
@@ -82,3 +82,38 @@ export const sortHistoryParametersByDate = (historyParameters: ParametersChange[
     return new Date(a.changeDate).valueOf() - new Date(b.changeDate).valueOf()
   })
 }
+
+export const sortParameterList = (parameters: ParameterConfig[]): ParameterConfig[] => {
+  const aggressivenessParameters = parameters.filter(parameter => parameter.unit === Unit.Percent)
+  const insulinParameters = parameters.filter(parameter => parameter.unit === Unit.InsulinUnit)
+  const mealParameters = parameters.filter(parameter => parameter.unit === Unit.Gram)
+  const weightParameters = parameters.filter(parameter => parameter.unit === Unit.Kilogram)
+  const thresholdParameters = parameters.filter(parameter => parameter.unit === Unit.MilligramPerDeciliter)
+
+  const sortedMealParameters = mealParameters.length > 0 ? sortMealParameters(mealParameters) : mealParameters
+  return [...insulinParameters, ...weightParameters, ...thresholdParameters, ...aggressivenessParameters, ...sortedMealParameters]
+}
+
+const sortMealParameters = (parameters: ParameterConfig[]): ParameterConfig[] => {
+  const breakfastMeals = []
+  const lunchMeals = []
+  const dinnerMeals = []
+  parameters.forEach((parameter) => {
+    const splitParameterKey = parameter.name.split('_')
+    switch (splitParameterKey[2]) {
+      case 'BREAKFAST':
+        breakfastMeals.push(parameter)
+        break
+      case 'LUNCH':
+        lunchMeals.push(parameter)
+        break
+      case 'DINNER':
+        dinnerMeals.push(parameter)
+        break
+    }
+  })
+  breakfastMeals[0] = breakfastMeals.splice(1, 1, breakfastMeals[0])[0]
```

Comment:
What is happening here? :rofl:

Reply:
I know this line is weird 😅 but it's here to sort the meal parameters in a logical order : small, medium and large. However it's sorted in another way: medium, small, large.